packages/cozy-procedures/src/redux/documentsDataSlice.js

```diff
@@ -25,59 +29,76 @@ const documentsSlice = createSlice({
       return {
         completedFromDrive: 0,
         data: Object.keys(action.payload).reduce((acc, fieldId) => {
+          //init files with undefined value
           acc[fieldId] = {
-            files: []
+            files: Array.from(Array(action.payload[fieldId].count))
           }
           return acc
         }, {}),
-        ui: Object.keys(action.payload).reduce((acc, fieldId) => {
-          acc[fieldId] = {
-            loading: false,
-            error: false
+        ui: Object.keys(action.payload).reduce(
+          (acc, fieldId) => {
+            const count = action.payload[fieldId].count
+            acc[fieldId] = Array.from(Array(count))
+            for (let i = 0; i < count; i++) {
+              acc[fieldId][i] = {
+                loading: false,
+                error: false
+              }
+            }
+
+            return acc
+          },
+          {
+            initiated: false
           }
-          return acc
-        }, {})
+        )
       }
     },
 
     fetchDocumentLoading: (state, action) => {
-      const { idDoctemplate, loading } = action.payload
-      state.ui[idDoctemplate].loading = loading
+      const { idDoctemplate, index } = action.payload
+      state.ui[idDoctemplate][index].loading = true
     },
     fetchDocumentSuccess: (state, action) => {
-      const { idDoctemplate, loading, files } = action.payload
-      state.ui[idDoctemplate].loading = loading
-      state.data[idDoctemplate].files = files
-      state.completedFromDrive += files.length
+      const { idDoctemplate, file, index } = action.payload
+      state.ui[idDoctemplate][index].loading = false
+      state.data[idDoctemplate].files[index] = file
+      state.completedFromDrive += 1
     },
     fetchDocumentError: (state, action) => {
-      const { idDoctemplate, error } = action.payload
-      state.ui[idDoctemplate].error = error
+      const { idDoctemplate, error, index } = action.payload
+      state.ui[idDoctemplate][index].error = error
+      state.ui[idDoctemplate][index].loading = false
     },
     setProcedureStatus: (state, action) => {
       state.ui.initiated = action.payload.initiated
     },
     unlinkDocument: (state, action) => {
-      const { document, documentId } = action.payload
-
-      state.data[documentId].files = state.data[documentId].files.filter(
-        doc => doc._id !== document.id
-      )
+      const { documentId, index } = action.payload
+      state.data[documentId].files[index] = undefined
```

Comment:
We don't need to change the state.ui here? `error` and `loading` don't need to be reset?